### PR TITLE
Initialize project.

### PR DESCRIPTION
**Pull request by kau-wtag**
In this branch I initialized a new rails project.

Background:
Starting a new project to develop event management system using Ruby on Rails. This project will allow users to manage events, including creating and attending. This is initial code base of a new project in rails.

Goal:

- Initialize the Rails Project.
- Update database.yml file directly to configure the database connection.

Technical Hints:

Acceptance:

- The Rails project is successfully created with the specified Rails version.
- Database configurations are updated in database.yml
- Rails server runs successfully.

Story link: [https://trello.com/c/ZBnhEAmq](https://trello.com/c/ZBnhEAmq)

### CODE REVIEW

File: config/credentials.yml.enc [hunk 1]
```diff
@@ -0,0 +1 @@
+flhsLfBuTMe+6sezqHvHt8mOGXzUnNZDxRBodExQY56oAlMeMd4vx/U8gz1DuGLQCglPDnFolcXqoP81eS6tdfuALNvXkuG1DY/3TuejuXjWYju+pi5QzeBV6kT7hI2eRymDxixERfRUHQBDSiD8pR/v282DH4u6idqInuJJHjHycI9c0Nce6Ucjcf4U4UhYhgFqhJ7FB02kZLWMXIRbOruGzUVrK98vO4bAGwipNzwc5iUTICH0rDLejXrIepzHcsslOJ4cyf6CFCuyOAJOencWn4+BRVGcgZlxjA6uKP7/pJebq/4kSIcHskkAfhIkabXI9Rc4nsvqadzCGhkOMG/1gpuO7on+E3ZlhMcFCTSZpbuU3NLBJYtNX4ofZ3rX96R+eUEMznkQUe9BthYA0D+PJ4qe--RhYis1TnFa8dR8oy--Hod6JNYOfoJTx8SRqVrDAQ==
```
Comment (asis-wtag) [inline config/credentials.yml.enc:1]: add a new line at the end of the file to remove the github warning

**Review by asis-wtag — APPROVED** [review]

File: .dockerignore
Comment (sal-wtag) [inline .dockerignore:1]: @udk-wtag bhai, are you using `Docker`? If not, I think it is better to use all unused/unnecessary files.

File: .gitattributes
Comment (sal-wtag) [inline .gitattributes:1]: @udk-wtag bhai, what is the purpose of this file?

File: Dockerfile
Comment (sal-wtag) [inline Dockerfile:1]: @udk-wtag bhai, ditto.

File: Gemfile
Comment (sal-wtag) [inline Gemfile:1]: @udk-wtag bhai, Gems should be sorted in alphabetical order.

File: app/javascript/controllers/hello_controller.js
Comment (sal-wtag) [inline app/javascript/controllers/hello_controller.js:1]: @udk-wtag bhai, is not this file unnecessary?

File: config/database.yml
Comment (sal-wtag) [inline config/database.yml:1]: @udk-wtag bhai, I think storing credentials (e.g. username and password) inside a `.env` file and then referring to them from the actual file, in this case `database.yml` is a better approach.

**Review by sal-wtag — COMMENTED** [review]
@udk-wtag Nice work.

## Few suggestions

- Setup `RSpec` instead of the default testing framework.
- Remove unnecessary files and Gems.
- Use `.env` file.

Thank You.

File: config/credentials.yml.enc (hunk 1 above)
Comment (kau-wtag) [reply config/credentials.yml.enc:1]: @asis-wtag If I add a new line, rails server is not starting. It gives some error.

File: .dockerignore
Comment (kau-wtag) [reply .dockerignore:1]: @sal-wtag bhai, Removed docker files.

File: Dockerfile
Comment (kau-wtag) [reply Dockerfile:1]: Bhai, removed this file too.

File: app/javascript/controllers/hello_controller.js
Comment (kau-wtag) [reply app/javascript/controllers/hello_controller.js:1]: yeah, This file is unnecessary. Removed it.

File: config/database.yml
Comment (kau-wtag) [reply config/database.yml:1]: yeah bhai, added `.env` file and stored all credentials into it.

File: Gemfile
Comment (sal-wtag) [inline Gemfile:1]: @udk-wtag bhai, we should remove commented out or unnecessary gems from the `Gemfile` and try to keep it tidy.

**Review by sal-wtag — APPROVED** [review]

File: Gemfile
Comment (kau-wtag) [reply Gemfile:1]: @sal-wtag  Ok bhai.

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+source 'https://rubygems.org'
+
+ruby '3.3.1'
+
+gem 'bootsnap', require: false
+gem 'dotenv-rails', groups: %i[development test]
```
Comment (SKH-WTAG) [inline Gemfile]: What about production?

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+gem 'importmap-rails'
+gem 'jbuilder'
```
Comment (SKH-WTAG) [inline Gemfile]: why do we need jbuilder?

File: Gemfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+gem 'pg', '~> 1.1'
+gem 'puma', '>= 5.0'
+gem 'rails', '~> 7.1.3', '>= 7.1.3.4'
+gem 'sprockets-rails'
+gem 'stimulus-rails'
+gem 'turbo-rails'
+gem 'tzinfo-data', platforms: %i[windows jruby]
+
+group :development, :test do
+  gem 'rspec-rails'
```
Comment (SKH-WTAG) [inline Gemfile:17]: Do we need rspec in development?

File: Gemfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+end
+
+group :development do
+  gem 'web-console'
+end
+
+group :test do
+  gem 'capybara'
+  gem 'selenium-webdriver'
```
Comment (SKH-WTAG) [inline Gemfile]: Why do we need this?

File: bin/setup [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  # Add necessary setup steps to this file.
+
+  puts "== Installing dependencies =="
+  system! "gem install bundler --conservative"
+  system("bundle check") || system!("bundle install")
+
+  # puts "\n== Copying sample files =="
+  # unless File.exist?("config/database.yml")
+  #   FileUtils.cp "config/database.yml.sample", "config/database.yml"
+  # end
```
Comment (SKH-WTAG) [inline bin/setup]: Uncomment these lines and create `config/database.yml.sample` file.

File: .rspec [hunk 7]
```diff
@@ -0,0 +1 @@
+--require spec_helper
```
Comment (SKH-WTAG) [inline .rspec]: Why do we need this?

File: Gemfile (hunk 3 above)
Comment (kau-wtag) [reply Gemfile]: @SKH-WTAG bhai,  We use `jbuilder` for creating JSON for APIs. also we can define JSON structure in a ruby-like syntax using it. We dont have to manually construct JSON in controllers or models.

File: Gemfile (hunk 2 above)
Comment (kau-wtag) [reply Gemfile]: @SKH-WTAG In production, we set environment variables directly on the server or deployment process. That's why we are not including production in the group.

File: Gemfile (hunk 4 above)
Comment (kau-wtag) [reply Gemfile:17]: @SKH-WTAG bhai, Though we don't need `rspec` in development, but if we do use in development it slightly increase load time in development environment, And without using `rspec` in the development rails generator won't automatically create corresponding `rspec` test files for our models, controllers etc. We have to manually create test files.

File: Gemfile (hunk 5 above)
Comment (kau-wtag) [reply Gemfile]: @SKH-WTAG bhai, we need `capybara` to write test files that helps to simulate how a user would interact with web app. It provides domain-specific-language for writing tests. The `selenium-webdriver` is a automating tool for web browser. It allows `capybara` to run tests in real browser environment.

File: bin/setup (hunk 6 above)
Comment (kau-wtag) [reply bin/setup]: @SKH-WTAG bhai, Done.

File: .rspec (hunk 7 above)
Comment (kau-wtag) [reply .rspec]: @SKH-WTAG bhai, The --require spec_helper line specifies that before running any tests, `spec_helper.rb` file should load. the `spec_helper.rb` contains configuration needed for all tests.

File: Gemfile
Comment (kau-wtag) [reply Gemfile:1]: bhai, Done.

File: .rspec (hunk 7 above)
Comment (SKH-WTAG) [reply .rspec]: In `rails_helper.rb` file, you are requiring this file. Do we need this?

File: Gemfile (hunk 2 above)
Comment (SKH-WTAG) [reply Gemfile]: We already have development and test group. Can we move it there?

File: Gemfile (hunk 3 above)
Comment (SKH-WTAG) [reply Gemfile]: AFAIK, currently we are using Rails View and not exposing any APIs.

For APIs, we can use Grape API or JBuilder. So, we can add this when we need.

File: Gemfile (hunk 5 above)
Comment (SKH-WTAG) [reply Gemfile]: For now, we won't test views using browser. We can add this later if needed.

File: config/database.yml [hunk 8]
```diff
@@ -0,0 +1,85 @@
+# PostgreSQL. Versions 9.3 and up are supported.
```
Comment (SKH-WTAG) [inline config/database.yml]: ignore this file. Because we have sample config

File: Gemfile (hunk 2 above)
Comment (kau-wtag) [reply Gemfile]: @SKH-WTAG yeah bhai, we can move it there and I have moved it.

File: Gemfile (hunk 3 above)
Comment (kau-wtag) [reply Gemfile]: @SKH-WTAG bhai, Removed jbuilder.

File: Gemfile (hunk 5 above)
Comment (kau-wtag) [reply Gemfile]: @SKH-WTAG bhai, Removed capybara and selenium-wevdriver.

File: config/database.yml (hunk 8 above)
Comment (kau-wtag) [reply config/database.yml]: @SKH-WTAG bhai, done.

File: .rspec (hunk 7 above)
Comment (kau-wtag) [reply .rspec]: @SKH-WTAG bhai, file removed.

File: .gitattributes
Comment (kau-wtag) [reply .gitattributes:1]: bhai, ok.

**Review by SKH-WTAG — APPROVED** [review]

**Pull request merged**